spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -270,6 +282,8 @@ && readSchema().equals(that.readSchema())
         && Objects.equals(startSnapshotId, that.startSnapshotId)
         && Objects.equals(endSnapshotId, that.endSnapshotId)
         && Objects.equals(asOfTimestamp, that.asOfTimestamp);
+    //        && Objects.equals(branch, that.branch)
+    //        && Objects.equals(tag, that.tag);
   }
```

Comment:
have to uncomment this , but a checkstyle cyclomatic complexity error.

Reply:
Considering it's required for a correct equals implementation of SparkBatchQueryScan, I think it makes the most sense just to suppress the warnings on the method  `@SuppressWarnings("checkstyle:CyclomaticComplexity")`